internal_ws/yktrace/src/tir.rs

```diff
@@ -153,10 +154,23 @@ impl<'a, 'm> TirTrace<'a, 'm> {
                             idx: rnm.rename_irplace(idx, &body),
                             scale: *scale,
                         },
-                        Statement::Store(dst, src) => Statement::Store(
-                            rnm.rename_irplace(dst, &body),
-                            rnm.rename_irplace(src, &body),
-                        ),
+                        Statement::Store(dst, src) => {
+                            if matches!(dst, IRPlace::Val { local, .. } if local == &sir::RETURN_LOCAL)
+                            {
+                                if matches!(src, IRPlace::Const { val: Constant::Bool(false), ..}) {
+                                    if body.flags.contains(BodyFlags::INTERP_STEP) {
+                                        // If the `interp_step` function returns false, we enabled
+                                        // trace stitching which loops the trace indefinitely until
+                                        // a guard fails.
+                                        stitch_trace = true;
+                                    }
+                                }
```

Comment:
does there need to be an `else { unreachable! }` around here somewhere for the case that we are in the intep_step, but the type isn't what we expect?

Reply:
I guess we can. Though we already check that the return type is `bool` at compile time, so another check here isn't really needed.

Reply:
I think we should, as a sanity check. In theory no "unreachable!" should ever be needed, but we insert them to check our understanding of things.

Reply:
I tend to agree with @vext01 -- it means if/when we change another bit of code elsewhere we are more likely to have a nice warning that we've broken a long-forgotten invariant.

Reply:
I'm not sure I understand.

Reply:
Fixed in d7f02a4.